src/client/Microsoft.Identity.Client/AppConfig/ManagedIdentityApplicationBuilder.cs

```diff
@@ -47,92 +47,51 @@ internal ManagedIdentityApplicationBuilder(ApplicationConfiguration configuratio
-        {
-            ApplicationBase.GuardMobileFrameworks();
-
-            var config = new ApplicationConfiguration(MsalClientType.ManagedIdentityClient);
-            return new ManagedIdentityApplicationBuilder(config)
-                .WithCacheSynchronization(false);
-        }
-
         /// <summary>
         /// Creates a ManagedIdentityApplicationBuilder from a user assigned managed identity clientID / resourceId.
-        /// See https://aka.ms/msal-net-managed-identity
+        /// For example, for a system assigned managed identity use ManagedIdentityApplicationBuilder.Create(SystemAssignedManagedIdentity.Default())
+        /// and for a user assigned managed identity use ManagedIdentityApplicationBuilder.Create(UserAssignedManagedIdentity.FromClientId(clientId)).
+        /// For more details see https://aka.ms/msal-net-managed-identity
         /// </summary>
-        /// <param name="userAssignedId">Client ID / Resource ID of the user assigned managed identity assigned to the resource.</param>
+        /// <param name="managedIdentityConfiguration">Configuration of the Managed Identity assigned to the resource.</param>
         /// <returns>A <see cref="ManagedIdentityApplicationBuilder"/> from which to set more
         /// parameters, and to create a managed identity application instance</returns>
 #if !SUPPORTS_CONFIDENTIAL_CLIENT
         [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide confidential client on mobile
 #endif
-        public static ManagedIdentityApplicationBuilder Create(string userAssignedId)
+        public static ManagedIdentityApplicationBuilder Create(ManagedIdentityConfiguration managedIdentityConfiguration)
         {
             ApplicationBase.GuardMobileFrameworks();
 
-            if (string.IsNullOrWhiteSpace(userAssignedId))
-            {
-                throw new ArgumentNullException(nameof(userAssignedId));
-            }
-
-            var config = new ApplicationConfiguration(MsalClientType.ManagedIdentityClient);
-            return new ManagedIdentityApplicationBuilder(config)
-                .WithUserAssignedManagedIdentity(userAssignedId)
-                .WithCacheSynchronization(false);
+            return new ManagedIdentityApplicationBuilder(BuildConfiguration(managedIdentityConfiguration));
         }
 
-        private ManagedIdentityApplicationBuilder WithUserAssignedManagedIdentity(string userAssignedId)
+        private static ApplicationConfiguration BuildConfiguration(ManagedIdentityConfiguration managedIdentityConfiguration)
         {
-            Config.IsUserAssignedManagedIdentity = true;
+            _ = managedIdentityConfiguration ?? throw new ArgumentNullException(nameof(managedIdentityConfiguration));
+            var config = new ApplicationConfiguration(MsalClientType.ManagedIdentityClient);
 
-            if (Guid.TryParse(userAssignedId, out _))
+            switch (managedIdentityConfiguration.IdType)
```

Comment:
I feel like this switch statement can be simplified to an if statement with some duplicate lines removed. For example `config.IsUserAssignedManagedIdentity = true;` probably doesnt need to be in 2 places